src/main/java/org/cactoos/http/HtBody.java

```diff
@@ -60,8 +59,117 @@ public HtBody(final Input rsp) {
 
     @Override
     public InputStream stream() throws IOException {
-        return new InputOf(
-            new TextOf(this.response).asString().split("\r\n\r\n")[1]
-        ).stream();
+        return new SkipInput(this.response, new BytesOf("\r\n\r\n")).stream();
+    }
+
+    /**
+     * {@link Input} that skips until it find some defined bytes.
+     *
+     * @author Victor Noel (victor.noel@crazydwarves.org)
+     * @version $Id$
+     * @since 0.1
+     */
+    static final class SkipInput implements Input {
+
+        /**
+         * The input.
+         */
+        private final Input origin;
+
+        /**
+         * The delimiter.
+         */
+        private final Bytes delimiter;
+
+        /**
+         * Ctor.
+         *
+         * @param origin The input.
+         * @param delimiter The bytes delimiter to skip until.
+         */
+        SkipInput(final Input origin, final Bytes delimiter) {
+            this.origin = origin;
+            this.delimiter = delimiter;
+        }
+
+        @Override
+        public InputStream stream() throws IOException {
+            final byte[] bytes = this.delimiter.asBytes();
+            final BoundedByteBuffer buffer = new BoundedByteBuffer(
+                bytes.length
+            );
+            final InputStream stream = this.origin.stream();
+            boolean eof = false;
+            while (!eof && !buffer.isEqualsTo(bytes)) {
+                final int read = stream.read();
+                if (read < 0) {
+                    eof = true;
+                } else {
+                    buffer.offer((byte) read);
+                }
+            }
+            return stream;
+        }
+    }
+
+    /**
+     * A very simple circular buffer of bytes.
+     *
+     * @author Victor Noel (victor.noel@crazydwarves.org)
+     * @version $Id$
+     * @since 0.1
+     */
+    static final class BoundedByteBuffer {
+
+        /**
+         * The buffer.
+         */
+        private final Queue<Byte> internal;
```

Comment:
@victornoel Huge overhead. `Byte` take 16 bytes of RAM per instance. `LinkedList` also adds overhead about 24 bytes per element. So in the result, you have `40x` overhead here. As I see, you do it because of circular, but see how `ArrayDeque` works.

Reply:
@SharplEr good point, ArrayDeque is clearly adequate for the use. I also added a `@todo` to make this even better in the future.